test_ffi/tests/test.js

```diff
@@ -454,6 +466,36 @@ logManyParametersFast(255, 65535, 4294967295, 4294967296, 123.456, 789.876, -1,
 logManyParametersFast(255, 65535, 4294967295, 4294967296, 123.456, 789.876, -1, -2, -3, -4, -1000, 1000, 12345.678910, 12345.678910, 12345.678910, 12345.678910, 12345.678910, 12345.678910, 12345.678910);
 assertFastCall(logManyParametersFast);
 
+function castU8U32Fast(x) { return symbols.cast_u8_u32(x); };
+
+%PrepareFunctionForOptimization(castU8U32Fast);
+console.log(castU8U32Fast(256));
+%OptimizeFunctionOnNextCall(castU8U32Fast);
+// In SysV AMD64 compilers rely on the convention to zero/sign-extend arguments by the caller to optimize the callee function.
+// If the trampoline did not zero/sign-extend arguments, this would return 256 instead of the expected 0 (in optimized builds)
+console.log(castU8U32Fast(256));
+assertFastCall(castU8U32Fast);
+
+function castU32U8Fast(x) { return symbols.cast_u32_u8(x); };
+
+%PrepareFunctionForOptimization(castU32U8Fast);
+console.log(castU32U8Fast(256));
+%OptimizeFunctionOnNextCall(castU32U8Fast);
+// In SysV AMD64 compilers rely on the convention to expect garbage in the bits beyond the size of the return value to optimize the callee function.
+// If the trampoline did not zero/sign-extend the return value, this would return 256 instead of the expected 0 (in optimized builds)
+console.log(castU32U8Fast(256));
+assertFastCall(castU32U8Fast);
+
+function add10U8Fast(a, b, c, d, e, f, g, h, i, j) { return symbols.add_10_u8(a, b, c, d, e, f, g, h, i, j); };
+
+%PrepareFunctionForOptimization(add10U8Fast);
```

Comment:
Consider making this into an assert helper function like this:

```ts
function assertOptimizes(fn, callback) {
  &PrepareFunctionForOptimization(fn);
  console.log(callback());
  %OptimizeFunctionOnNextCall(fn);
  console.log(callback());
  assertFastCall(fn);
}

// SysV ABI comment
assertOptimizes(add10U8Fast, () => add10U8Fast(0, 1, 2, 3, 4, 5, 6, 7, 8, 9));
```

Reply:
Yeah, saw your suggestion in Discord, I'll definitely apply it.

Reply:
What do you think about naming this function `testOptimized` instead of `assertOptimizes`? The reason is that It also tests the function itself, not only if it has been optimized.